src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/ImmutableDispatchRuntime.cs

```diff
@@ -660,6 +682,50 @@ internal async Task<MessageRpc> ProcessMessageAsync(MessageRpc rpc)
             return rpc;
         }
 
+        private Activity CreateActivity(ref Message request, IClientChannel channel, InstanceContext instanceContext)
+        {
+            var activity = WcfInstrumentationActivitySource.ActivitySource.StartActivity(
+                WcfInstrumentationActivitySource.IncomingRequestActivityName,
+                ActivityKind.Server);
+
+            if (activity != null)
+            {
+                string action;
+                if (!string.IsNullOrEmpty(request.Headers.Action))
+                {
+                    action = request.Headers.Action;
+                    activity.DisplayName = action;
+                }
+                else
+                {
+                    action = string.Empty;
+                }
+
+                if (activity.IsAllDataRequested)
+                {
+                    activity.SetTag(WcfInstrumentationConstants.RpcSystemTag,
+                        WcfInstrumentationConstants.WcfSystemValue);
+
+                    var actionMetadata = GetActionMetadata(request, action);
+
+                    activity.SetTag(WcfInstrumentationConstants.RpcServiceTag, actionMetadata.ContractName);
```

Comment:
whould we set `RpcServiceTag` and `RpcMethodTag` with null values ?

Reply:
'RpcServiceTag' could be null --> should probably only write it when set,
'RpcMethodTag' should not be null --> string.Empty, or the Action from the request Header. So should normally be set in most cases